docs/registering-strategies.adoc

```diff
@@ -0,0 +1,80 @@
+== Registering Custom Strategies
+
+By default *Smart Testing* comes with a set of strategies such as `new`, `changed`, `affected` or `failed`, but you can implement your own strategies and registering them.
+
+To do it you just need to follow next four steps:
+
+* Implement `org.arquillian.smart.testing.spi.TestExecutionPlannerFactory` Java SPI and registering as such.
+* Implement `org.arquillian.smart.testing.spi.TestExecutionPlanner` interface.
+* Register strategy on Smart Testing configuration file (optional).
+
+=== Implementing `TestExecutionPlannerFactory`
+
+`TestExecutionPlannerFactory` is a Java SPI interface that manages the creation of the strategy.
+You need to implement three simple methods, two related with the name of the strategy and one that generates the strategy class.
```

Comment:
While reading this - isn't a need for implementing two methods for the name somewhat a smell? @lordofthejars as you were looking at ability to bring those custom strategies - do you see any potential improvements so that we can have lean, but powerful mechanism in place?

Reply:
I thought exactly the same, but since I didn't have the knowledge on why it was in this way, I decided that for this PR leave it as this and discuss in another PR. Notice that one is the real name and the other is the alias.

Reply:
All knowledge is in the code ;)

Reply:
But anyway, the purpose of this PR is not simplify strategies "architecture" but allowing user to register its custom strategies. I think this should go to another PR.

Reply:
I didn't really ask for changes here - I wanted to know if you gave it some thought.

Reply:
yes, I did it, and it seems there are some redundant info.